msal/src/Microsoft.Identity.Client/Features/ConfidentialClient/ConfidentialClientApplication.cs

```diff
@@ -318,7 +318,7 @@ async Task<AuthenticationResult> IConfidentialClientApplicationWithCertificate.A
             requestParameters.ExtraQueryParameters = extraQueryParameters;
 
             var handler =
-                new InteractiveRequest(HttpManager, CryptographyManager, requestParameters, null, loginHint, UIBehavior.SelectAccount, null);
+                new InteractiveRequest(HttpManager, CryptographyManager, requestParameters, ApiEvent.ApiIds.None, null, loginHint, UIBehavior.SelectAccount, null);
```

Comment:
I wonder why ApiId is null / none in this case. Is it because we do not / should not have telemetry for confidential client?

Reply:
no, we should have telemetry for confidential clients as well.

Reply:
Only because it wasn't there before.  I kept it this way for the first round PR to explicitly raise this issue to ensure we realize the gaps in our telemetry.  I'll see if we already have a code for this and fix any of the null/none cases.

Reply:
Just looked.  We don't have an API ID for GetAuthorizationRequestUrlAsync() and those are the two cases in the public API where API ID is null.  I'm going to file a separate issue for that so I don't hold up this PR since it's not regressing existing functionality.